src/ns-migration.sol

```diff
@@ -0,0 +1,190 @@
+// - add & wire mkr adapter contracts: clerk & mgr, spotter, vat
+contract TinlakeSpell {
+
+    bool public done;
+    string constant public description = "Tinlake NS2 migration kovan Spell";
+
+    address constant public ROOT = 0xB86a5F04F5aE402d8547842A29071309C23b47e1;
+    address constant public SHELF = 0xe03A4812E1Ae31567932BDBf3618Ba5F64918bCB;
+    address constant public COLLECTOR = 0x751FF4Dd537aE8f1bA226Ae5d667a6406a53754C;
+    address constant public SENIOR_TOKEN = 0xa1ff44b597549aE3f79Ab02109DD84B465CD7cCC;
+    address constant public SENIOR_TRANCHE = 0xb3A6758FB6F79905692B3D805786B362cACB9f75;
+    address constant public SENIOR_MEMBERLIST = 0x71F515E53a9B0831c0Df03bcD33b6e6701F9e278;
+    address constant public JUNIOR_TRANCHE = 0xb148DA9cB8F7Ee64290105888D21aa3325681a6f;
+    address constant public ASSESSOR_WRAPPER = 0x7048AFBa9aBA4d3CF206e341bD22c2CA47061Fe8;
+    address constant public TINLAKE_CURRENCY = 0x99E21e1e7D99d06F780666A3BE6Ba178De04B0a9;
+    address constant public NAV = 0x8f90432c37d58aB79802B31e15F59556236123dA;
+
+    // contracts to be swapped
+    address constant public ASSESSOR_OLD  = 0x00423Eb98c9CC2F43080A3A3Df3bf2a58Fcf29EB;
+    address constant public COORDINATOR_OLD = 0xb0AefE053F73bEd8b922da60451f2eaEC71F4b2f;
+    address constant public RESERVE_OLD = 0xbCd87D68A7829A9c6B4C763D2e9F9f345c61E0f6;
+    address constant public COORDINATOR_NEW = 0x0000000000000000000000000000000000000000;
+    address constant public ASSESSOR_NEW  = 0x0000000000000000000000000000000000000000;
+    address constant public RESERVE_NEW = 0x0000000000000000000000000000000000000000;
+
+    // adapter contracts
+    address constant public CLERK  = 0x0000000000000000000000000000000000000000;
+    address constant public MGR = 0x0000000000000000000000000000000000000000;
+    address constant public SPOTTER = 0x0000000000000000000000000000000000000000;
+    address constant public VAT = 0x0000000000000000000000000000000000000000;
+
+    uint constant public CLERK_BUFFER = 0;
+
+    // permissions to be set
+    function cast() public {
+        require(!done, "spell-already-cast");
+        done = true;
+        execute();
+    }
+
+    function execute() internal {
+        TinlakeRootLike root = TinlakeRootLike(ROOT);
+        address self = address(this);
+        // set spell as ward on the core contract to be able to wire the new contracts correctly
+        root.relyContract(SHELF, self);
+        root.relyContract(COLLECTOR, self);
+        root.relyContract(ASSESSOR_NEW, self);
+        root.relyContract(COORDINATOR_NEW, self);
+        root.relyContract(RESERVE_NEW, self);
+        root.relyContract(CLERK, self);
+        // contract migration --> assumption: root contract is already ward on the new contracts
+        migrateAssessor();
+        migrateCoordinator();
+        migrateReserve();
+        integrateAdapter();
+    }
+
+
+    function migrateAssessor() internal {
```

Comment:
I am wondering if we could run into any security issues if we are not `deny` the old contract assessor contract

Reply:
we need to run it, but not here as we are also swapping the reserve